test/powershell/Language/Classes/Scripting.Classes.Attributes.Tests.ps1

```diff
@@ -216,3 +218,96 @@ Describe 'Type resolution with attributes' -Tag "CI" {
         }
     }
 }
+
+Describe 'ValidateSet support a dynamically generated set' -Tag "CI" {
+    Context 'C# test' {

```

Comment:
Following sample don't work - the type passed as string to `public ValidateSetAttribute(params string[] validValues)` constructor - wrong binding 😕 :
```powershell
            class GenValuesForParam2 : System.Management.Automation.IValidateSetValuesGenerator {
                [System.Collections.Generic.IEnumerable[String]] GetValidValues() { return "Test1","TestString","Test2" }
            }

            function Get-TestValidateSet3
            {
                [CmdletBinding()]
                Param
                (
                    [ValidateSet([GenValuesForParam2])]
                    $Param1
                )

                $Param1
            }
```